src/raps/actions/unlock.ts

```diff
@@ -112,15 +114,32 @@ const unlock = async (
     throw e;
   }
   let approval;
-  let gasPrice;
+  let maxFeePerGas;
+  let maxPriorityFeePerGas;
   try {
     // approvals should always use fast gas or custom (whatever is faster)
-    gasPrice = selectedGasPrice?.value?.amount;
-    const fastPrice = get(gasPrices, `[${gasUtils.FAST}].value.amount`);
-    if (greaterThan(fastPrice, gasPrice)) {
-      gasPrice = fastPrice;
+    maxFeePerGas = selectedGasFee?.gasFeeParams?.maxFeePerGas?.amount;
+    maxPriorityFeePerGas =
+      selectedGasFee?.gasFeeParams?.maxPriorityFeePerGas?.amount;
+
+    // if swap isn't the last action, use fast gas or custom (whatever is faster)
```

Comment:
looks like line 125-130 is copy pasta - it is not needed here bc unlock should always be done on fast

Reply:
whoa